src/coreclr/gc/gcpriv.h

```diff
@@ -5105,20 +5105,24 @@ size_t& generation_allocated_since_last_pin (generation* inst)
 #endif //FREE_USAGE_STATS
 
 inline
-float generation_allocator_efficiency (generation* inst)
+uint64_t generation_allocator_efficiency_percent (generation* inst)
 {
-    if ((generation_free_list_allocated (inst) + generation_free_obj_space (inst)) != 0)
-    {
-        return ((float) (generation_free_list_allocated (inst)) / (float)(generation_free_list_allocated (inst) + generation_free_obj_space (inst)));
-    }
-    else
-        return 0;
+    uint64_t free_obj_space = generation_free_obj_space (inst);
+    uint64_t free_list_allocated = generation_free_list_allocated (inst);
+    if (free_list_allocated==0)
+      return 0;
+    return (100 * free_list_allocated) / (free_list_allocated + free_obj_space);
 }
+
 inline
-size_t generation_unusable_fragmentation (generation* inst)
-{
-    return (size_t)(generation_free_obj_space (inst) +
-                    (1.0f-generation_allocator_efficiency(inst))*generation_free_list_space (inst));
+uint64_t generation_unusable_fragmentation (generation* inst)
+{
+    uint64_t free_obj_space = generation_free_obj_space (inst);
+    uint64_t free_list_allocated = generation_free_list_allocated (inst);
+    uint64_t free_list_space = generation_free_list_space (inst);
+    if (free_obj_space==0)
```

Comment:
```suggestion
    if ((free_list_allocated + free_obj_space) == 0)
```

Reply:
I think the idea is that if free_obj_space is zero, then the computation will be zero regardless of the free_list_allocated value.  (unless one can be negative...)  It's subtle and perhaps not worth the microoptimization or worth a comment.  The same patterns occurs in the efficiency method.

Reply:
Yes if `free_obj_space` is zero then the result is always zero. The operands are of type `size_t` and so can not be negative.

Reply:
yes, I got the idea. but this kind of microoptimization is not worth it at all. I'd much rather have the code be easier to understand.

Reply:
Thanks for your report. I will do as requested.